lmdeploy/serve/openai/api_server.py

```diff
@@ -416,11 +416,17 @@ async def chat_completions_v1(request: ChatCompletionRequest,
     - presence_penalty (replaced with repetition_penalty)
     - frequency_penalty (replaced with repetition_penalty)
     """
-    VariableInterface.session_id += 1
-    request.session_id = VariableInterface.session_id
+    if request.session_id == -1:
+        VariableInterface.session_id += 1
+        request.session_id = VariableInterface.session_id
     error_check_ret = await check_request(request)
     if error_check_ret is not None:
         return error_check_ret
+    if VariableInterface.async_engine.id2step.get(str(request.session_id),
```

Comment:
nit: Use `in` to determine whether the key exists in the `dict`. Will the readability be better? Is there a difference in performance compared to `get`? What is your suggestion? @AllentDan

Reply:
I just set the step to 0 instead of popping it out from `id2step` in `generate` func

Reply:
ok